### PR TITLE
Harden the make check script and provide more output for errors.

### PR DESCRIPTION
**Pull request by awallace-cray**
Adds more error checking and verbose diagnostics to
util/test/checkChplInstall, the shell script that implements
"make check".
    
* Set env variable CHPL_CHECK_DEBUG=1 everywhere make check is
  called from the nightly build scripts, to activate more verbose
  error diagnostics.
* The new verbose diagnostic messages are marked [Debug], and include:
  * Msgs at beginning and end of the make check script, including
    the exit status returned by the script to "make".
  * Exit status returned to the script from both the Chpl compilation
    and the test run.
  * Full copies of reference and actual test outputs, not just "diff".
* Explicitly check ("cd" to) all important file system locations,
  to guard against unexpected file system problems.

### CODE REVIEW

File: util/test/checkChplInstall [hunk 1]
```diff
@@ -95,10 +96,16 @@ else
...
+( cd "${CHPL_DIR}" ) || { log_error "failed cd $CHPL_DIR" ; exit 1 ; }
 
 # Tmp directory for compile output
 TMP_TEST_DIR="$(mktemp -d ${CHPL_DIR}/chapel-test-XXXXX)"
+if [ -z "$TMP_TEST_DIR" ] ; then
+    log_error "failed mktemp -d ${CHPL_DIR}/chapel-test-XXXXX"
+    exit 1
+fi
 log_info "Temporary test job directory: ${TMP_TEST_DIR}"
+( cd $TMP_TEST_DIR ) || { log_error "failed cd $TMP_TEST_DIR" ; exit 1 ; }
```
Comment (ben-albrecht) [inline util/test/checkChplInstall]: For consistency,  this should be `"${TMP_TEST_DIR}"`

File: util/test/checkChplInstall [hunk 2]
```diff
@@ -95,10 +96,16 @@ else
     log_info "${CHPL_DIR} already exists. Using it."
     CHPL_DIR_REMOVE=0
 fi
+( cd "${CHPL_DIR}" ) || { log_error "failed cd $CHPL_DIR" ; exit 1 ; }
 
 # Tmp directory for compile output
 TMP_TEST_DIR="$(mktemp -d ${CHPL_DIR}/chapel-test-XXXXX)"
+if [ -z "$TMP_TEST_DIR" ] ; then
```
Comment (ben-albrecht) [inline util/test/checkChplInstall]: similarly, `"${TMP_TEST_DIR}"`

File: util/test/checkChplInstall [hunk 3]
```diff
@@ -189,40 +200,63 @@ else
...
 # Run compiled binary with parsed execution options
 (
-    cd ${TMP_TEST_DIR}
+    cd ${TMP_TEST_DIR} || { log_error "failed cd $TMP_TEST_DIR" ; exit 1 ; }
 
     EXEC_OPTS="$(cat ${TEST_DIR}/${TEST_JOB}.execopts)"
 
     log_info "Running test job."
-    ./${TEST_JOB} -nl${NUMLOCALES} ${EXEC_OPTS} | sort > ${TEST_EXEC_OUT} 2>&1
+    ./${TEST_JOB} -nl${NUMLOCALES} ${EXEC_OPTS} | sort > ${TEST_EXEC_OUT}
```
Comment (ben-albrecht) [inline util/test/checkChplInstall]: Why drop stderr redirection? Is this more consistent with `sub_test`?

Comment (awallace-cray) [inline util/test/checkChplInstall]: Yes. Even if not, the only thing on stderr right there would be errors from "sort" itself. 
A better question is, why isn't the stderr from running the test job redirected into THAT stdout stream? It probably should be. But, that would NOT be consistent with sub_test. And, I'm not about to change sub_test.

Comment (ben-albrecht) [inline util/test/checkChplInstall]: OK makes sense

**Comment by awallace-cray** [comment]
How's that?

**Pull request merged**